bokeh/embed.py

```diff
@@ -131,10 +132,37 @@ def components(models, wrap_script=True, wrap_plot_info=True):
             values.append(models[k])
         models = values
 
-    # 2) Do our rendering
-
-    with _ModelInDocument(models):
-        (docs_json, render_items) = _standalone_docs_json_and_render_items(models)
+    # 2) Append models to one document. Either pre-existing or new.
+    doc = None
+    for model in models:
+        if isinstance(model, Document):
+            doc = model
+            break
+        elif isinstance(model, Model):
+            if model.document is not None:
+                doc = model.document
+                break
+    if doc is None:
+        # oh well - just make up a doc
+        doc = Document()
```

Comment:
I think we should continue to allow one document as long as none of the models have a different document. I guess we should change `models`; what would be a better name?